src/libcore/slice.rs

```diff
@@ -1183,6 +1183,34 @@ impl<'a, T> Iterator for Windows<'a, T> {
             (size, Some(size))
         }
     }
+
+    #[inline]
+    fn count(self) -> usize {
+        self.size_hint().0
+    }
+
+    #[inline]
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        let end = self.size + n;
```

Comment:
This addition may overflow. The methods for add with overflow might be appropriate, take the `None` case on overflow.